kv/memberlist/dnsprovider.go

```diff
@@ -0,0 +1,18 @@
+package memberlist
+
+import (
+	"context"
+)
+
+// DNSProvider supports storing or resolving a list of addresses.
+type DNSProvider interface {
+	// Resolve stores a list of provided addresses or their DNS records if requested.
+	// Addresses prefixed with `dns+` or `dnssrv+` will be resolved through respective DNS lookup (A/AAAA or SRV).
+	// For non-SRV records, it will return an error if a port is not supplied.
+	//
+	// Example: By providing the address of "dns+grafana.com:80"}, it should resolve into something like []string{"34.120.177.193:80", "[2600:1901:0:b3ea::]:80"}.
```

Comment:
@56quarters what do you think of this example? Couldn't come up with anything that produces results for `dnssrv+`.

Reply:
I think that's great to start.

Reply:
Oh, just saw @pstibrany's comment. That's fine too

Reply:
@56quarters sorry, we removed this documentation in the meantime since @pstibrany and @pracucci pointed out it's implementation specific :)